src/vs/workbench/contrib/debug/common/debugModel.ts

```diff
@@ -2032,6 +2033,10 @@ export class DebugModel extends Disposable implements IDebugModel {
 	removeWatchExpressions(id: string | null = null): void {
 		this.watchExpressions = id ? this.watchExpressions.filter(we => we.getId() !== id) : [];
 		this._onDidChangeWatchExpressions.fire(undefined);
+		if (!id) {
```

Comment:
It looks like if id is null, we removed _all_ expressions and can dispose all the listeners

Reply:
I think they will have been removed one by one? but sure, makes sense to add this